includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2343,4 +2464,161 @@ protected function guess_modal_role( DOMElement $modal ) {
 		// None of the roles we are looking for match any of the classes.
 		return Role::DIALOG;
 	}
+
+	/**
+	 * Evaluates the given XPath expression and give first Element if exist.
+	 *
+	 * @param string  $expression   The XPath expression to execute.
+	 * @param Element $context_node The optional context node can be specified for doing relative XPath queries.
+	 *                              By default, the queries are relative to the root element.
+	 *
+	 * @return Element|null Return Element if exists otherwise null.
+	 */
+	protected function get_first_element( $expression, $context_node = null ) {
+
+		$dom_node_list = $this->dom->xpath->query( $expression, $context_node );
+
+		/** @var Element $dom_node */
+		$dom_node = $dom_node_list->item( 0 );
+
+		return $dom_node;
+	}
+
+	/**
+	 * Update main navigation menu for "twentynineteen" theme.
+	 *
+	 * @return void
+	 */
+	public function update_twentynineteen_mobile_main_menu() {
+
+		$xpaths = [
+			'main_menu'               => '//nav[ @id = "site-navigation" ]//ul[ @class = "main-menu"]',
+			'top_menu_items'          => './li[ contains( @class, "menu-item" ) and contains( @class, "menu-item-has-children" ) ]',
+			'expand_button'           => './button[ @class = "submenu-expand" ]',
+			'submenu'                 => './ul[ @class = "sub-menu" ]',
+			'close_button_in_submenu' => './li[ contains( @class, "mobile-parent-nav-menu-item" ) ]//button[ contains( @class, "menu-item-link-return" ) ]',
+		];
+
+		$main_menu = $this->get_first_element( $xpaths['main_menu'] );
+		if ( empty( $main_menu ) ) {
+			return;
+		}
+
+		$menu_items = $this->dom->xpath->query( $xpaths['top_menu_items'], $main_menu );
+
+		/** @var Element $menu_item */
+		foreach ( $menu_items as $index => $menu_item ) {
```

Comment:
Nit: This variable isn't needed.

```suggestion
		foreach ( $menu_items as $menu_item ) {
```